presto-manager-common/src/main/java/com/teradata/prestomanager/common/NodeSet.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.common;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class NodeSet
+{
+    public static Set<Agent> nodeSet = ConcurrentHashMap.newKeySet();
+
+    private NodeSet()
+    {
+        //TODO: initialize the nodeMap, read from user input file
+    }
+
+    public static void addAgent(Agent agent)
+    {
+        nodeSet.add(agent);
+    }
+
+    public static void removeAgent(Agent agent)
+    {
+        nodeSet.remove(agent);
+    }
+
+    public static List<URI> getAllNodes()
+    {
+        return nodeSet.stream().map(Agent::getUri).collect(Collectors.toList());
+    }
+
+    public static URI getCoordinator()
+    {
+        return nodeSet.stream().filter(NodeSet::coordinatorFilter).map(Agent::getUri).collect(Collectors.toList()).get(0);
+    }
+
+    public static List<URI> getWorkers()
+    {
+        return nodeSet.stream().filter(NodeSet::workerFilter).map(Agent::getUri).collect(Collectors.toList());
+    }
+
+    private static boolean workerFilter(Agent agent)
+    {
+        return agent.getNodeRole() == NodeRole.WORKER;
+    }
+
+    private static boolean coordinatorFilter(Agent agent)
+    {
+        return agent.getNodeRole() == NodeRole.COORDINATOR;
+    }
+
+    public enum NodeRole
+    {
+        WORKER, COORDINATOR
+    }
+
+    //TODO: Add method to generate ID
+    public static class Agent
+    {
+        private URI uri;
+        private static int count;
```

Comment:
For now, I would stick with using the URL/IP as the ID, because we want it to be a bit more stable than the order they come online.

We can come up with something better later.